src/test/java/org/folio/rest/impl/EholdingsPackagesTest.java

```diff
@@ -0,0 +1,129 @@
+import io.restassured.specification.RequestSpecification;
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import org.apache.http.HttpStatus;
+import org.folio.config.cache.RMAPIConfigurationCache;
+import org.folio.rest.RestVerticle;
+import org.folio.rest.jaxrs.model.PackageCollection;
+import org.folio.rest.tools.utils.NetworkUtils;
+import org.folio.rest.util.RestConstants;
+import org.folio.util.TestUtil;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(VertxUnitRunner.class)
+public class EholdingsPackagesTest {
+
+  private static final String STUB_CUSTOMER_ID = "TEST_CUSTOMER_ID";
+  private static int port;
+  private static String host;
+
+  @org.junit.Rule
+  public WireMockRule userMockServer = new WireMockRule(
+    WireMockConfiguration.wireMockConfig()
+      .dynamicPort()
+      .notifier(new ConsoleNotifier(true)));
+
+  @BeforeClass
+  public static void setUpClass(final TestContext context) {
+    Vertx vertx = Vertx.vertx();
+    vertx.exceptionHandler(context.exceptionHandler());
+    port = NetworkUtils.nextFreePort();
+    host = "http://localhost";
+
+    DeploymentOptions restVerticleDeploymentOptions = new DeploymentOptions()
+      .setConfig(new JsonObject().put("http.port", port));
+    vertx.deployVerticle(RestVerticle.class.getName(), restVerticleDeploymentOptions,
+      context.asyncAssertSuccess());
+
+  }
+
+  @Before
+  public void setUp() {
+    RMAPIConfigurationCache.getInstance().invalidate();
+  }
+
+  @Test
+  public void shouldReturnPackagesOnGet() throws IOException, URISyntaxException {
+    String stubResponseFile = "responses/rmapi/packages/get-packages-response.json";
+
+    String wiremockUrl = host + ":" + userMockServer.port();
+    TestUtil.mockConfiguration("responses/configuration/get-configuration.json", wiremockUrl);
+    WireMock.stubFor(
+      WireMock.get(
+        new UrlPathPattern(new RegexPattern("/rm/rmaccounts/" + STUB_CUSTOMER_ID + "/packages.*"),
+          true))
+        .willReturn(new ResponseDefinitionBuilder()
+          .withBody(TestUtil.readFile(stubResponseFile))));
+
+    PackageCollection packages = RestAssured.given()
+      .spec(getRequestSpecification())
+      .when()
+      .get("eholdings/packages?q=American&filter[type]=abstractandindex&count=5")
+      .then()
+      .statusCode(HttpStatus.SC_OK).extract().as(PackageCollection.class);
+
+    comparePackages(packages);
+  }
+
+  @Test
+  public void shouldReturn400WhenCountInvalid() {
+    RestAssured.given()
+      .spec(getRequestSpecification())
+      .when()
+      .get("eholdings/packages?q=American&filter[type]=abstractandindex&count=500")
+      .then()
+      .statusCode(HttpStatus.SC_BAD_REQUEST);
+  }
+
```

Comment:
Could we add a test on the filter for custom packages (since it does the additional rm api vendor look up to support this)

Reply:
added